.github/workflows/codecov.yml

```diff
@@ -0,0 +1,42 @@
+name: Generate Codecov Report
+
+on:
+  push:
+    branches: [ develop, main ]
+  pull_request:
+    branches: [ develop, main ]
+
+jobs:
+  codecov_report:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        ref: ${{ github.head_ref }}
+        fetch-depth: 0
+          
+    - name: set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        java-version: '11'
+        distribution: 'adopt'
+        cache: gradle
+
+    - name: Run jacoco
+      run: |
+        ./gradlew agent:testReleaseUnitTestCoverage
+        ./gradlew agent-core:jacocoTestReport
+        ./gradlew instrumentation:jacocoTestReport
+        ./gradlew plugin:gradle:jacocoTestReport
+
+    - name: Upload Test Report Part 1
+      uses: codecov/codecov-action@v3
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+        files: '**/testReleaseUnitTestCoverage.xml'
+
```

Comment:
I couldn't combine paths for the files, separated into two steps

Reply:
That's fine, but the second jobs should be dependent on the first, is that correct? In that case I'd add a [needs statement](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds) statement to the second job.

Reply:
Actually they are all independents reports, doesn't reply on each other